src/sync/sync_session.cpp

```diff
@@ -636,8 +636,15 @@ void SyncSession::handle_error(SyncError error)
             }
             break;
         case NextStateAfterError::inactive: {
-            std::unique_lock<std::mutex> lock(m_state_mutex);
-            advance_state(lock, State::inactive);
+            if (error.is_client_reset_requested()) {
+                std::unique_lock<std::mutex> lock(m_state_mutex);
+                cancel_pending_waits(lock, make_error_code(realm::sync::ProtocolError::diverging_histories));
```

Comment:
Not sure I understand this. Are you faking an error code that result in an Resync? If I remember correctly there are multiple ones that triggers a Resync. Also, I thought the Client Resync would be transparent from the end user perspective? This seems to bubble up to them?

Reply:
I don't think the manual case should be fully transparent. As it is now, a manual client resync will lead to an `Operation canceled` while the proposed change will say `Divergent histories`. The SDKs can handle the non-manual cases transparent.

Reply:
No, the manual case needs to be handled for sure. But I'm still not sure why you are faking the error code because shouldn't you have the "correct" one somewhere?